src/api/routes/company.js

```diff
@@ -51,11 +55,48 @@ module.exports = (app) => {
         const computedOffset = parseInt(offset || 0, 10);
 
         try {
-            const { companies, totalDocCount } = await new CompanyService().findAll(computedLimit, computedOffset);
+            const { companies, totalDocCount } =
+                await new CompanyService().findAll(computedLimit, computedOffset, req.hasAdminPrivileges, req.hasAdminPrivileges);
             return res.json({ companies, totalDocCount });
         } catch (error) {
             return next(error);
         }
 
     });
+
+    router.put(
+        "/:companyId/block",
+        or([
+            authMiddleware.isGod,
+            authMiddleware.isAdmin
+        ],
+        { status_code: HTTPStatus.UNAUTHORIZED, error_code: ErrorTypes.FORBIDDEN, msg: ValidationReasons.INSUFFICIENT_PERMISSIONS }),
+        validators.block,
+        validators.canBlock,
```

Comment:
Now that I think about it, it shouldn't be PUT, since the company would be blocked for different reasons, does this make sense?

Reply:
Close, but not quite.

A PUT request represents an idempotent request. This can have many different meanings, but you can think of it this way: a request is idempotent if the application state after 1 call is the same as the state after n calls. The trivial case is where there are no side effects, it will obviously be idempotent. However, in this case, if you repeat the same request (including the args) n times, it shouldn't have any impact on the final result.

Think of changing the arg as executing a different operation. That operation should be reproducible without changing state after the first call, therefore it should be a PUT and there should not be validation as to whether the company is already blocked or not.

Reply:
source: https://en.wikipedia.org/wiki/Idempotence#Computer_science_examples